hasher-matcher-actioner/CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+Getting Started with Contributing
+
+In it's rough state, the prototype needs some prior configuration / setup before you are ready to start developing on it. Below are some steps and tips for getting started. If anything needs correction, PRs are always welcome.
+
+## Prerequisites
+
+In the prototype's current state, the following tools are prerequisites to deploying the HMA prototype.
+
+1. [terraform cli](https://www.terraform.io/)
+2. [Docker](https://www.docker.com/)
+
+Additionally, if you want to make use of the provided scripts for doing things like uploading new Docker images containing the code for the various AWS lambda functions used by HMA, you will want the following tools installed
+
+1. [jq](https://stedolan.github.io/jq/)
+2. [aws cli](https://aws.amazon.com/cli/)
+
+Beyond tooling you need to have access to an AWS account where the various resources defined in the terraform files here will be created. You will want to either have your AWS credentials either in your environment or in a centralized credentials file. (See the [aws terraform provider documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#authentication) for more information on these credentials) **Note: These applying these terraform files to your AWS account will result in resources being created that will create costs for your account.**
+
+## Config Files
+
+Before using terraform, you will need to provide some additional configuration, examples of which are provided in this folder using the same name, but with example suffixed to the end.
+
+1. `terraform.tfvars`: In this file, you will want to define the values for the variables defined in [variables.tf](variables.tf). Only variables without defaults are required, though if you are setting up an environment for developing on the HMA prototype, you likely want to override the default `prefix` to allow having an isolated environment. See the terraform docs on [input variables](https://www.terraform.io/docs/configuration/variables.html) for more information on providing variable values and overrides.
```

Comment:
One thing that threw me off while I was getting onboard was that you can't finish backend.tf until you've done the building of the image, but the order is reversed here.

I assume that this problem goes away if we can check in an image?

Reply:
Good point. Going to swap these two sections (config files and building and pushing the docker image).

And yes, once we have a production image that we are using, we can provide that as the default for the `hma_lambda_docker_uri` variable and then the order will not matter as much.